fenicsxprecice/fenicsxprecice.py

```diff
@@ -170,139 +188,151 @@ def write_data(self, write_function):
-            # Ensure that function spaces of read and write functions are defined using the same mesh
-            self._write_function_type = determine_function_type(write_function_space)
-            self._write_function_space = write_function_space
-
-        # Set vertices on the coupling subdomain for this rank
-        self._fenicsx_dims = function_space.mesh.geometry.dim
-        # returns 3d coordinates (necessary later for writing the data!)
-        ids, coords = get_fenicsx_vertices(function_space, coupling_subdomain, self._fenicsx_dims)
-        # this isnt a problem in update_coupling_expression, because in this function
-        # , the two first dimensions are extracted. Exactly what we want!
-        self._fenicsx_vertices.set_ids(ids)
-        self._fenicsx_vertices.set_coordinates(coords)
-
-        # Set up mesh in preCICE
-        self._precice_vertex_ids = self._participant.set_mesh_vertices(
-            self._config.get_coupling_mesh_name(), self._fenicsx_vertices.get_coordinates()[
-                :, :2])  # give preCICE only 2D coordinates
-
-        if self._fenicsx_vertices.get_ids().size > 0:
-            self._empty_rank = False
-        else:
-            print("Rank {} has no part of coupling boundary.".format(self._comm.Get_rank()))
-
-        # Ensure that function spaces of read and write functions use the same mesh
-        if self._coupling_type is CouplingMode.BI_DIRECTIONAL_COUPLING:
-            assert (self._read_function_space.mesh is write_function_space.mesh
-                    ), "read_function_space and write_object need to be defined using the same mesh"
-
-        if self._fenicsx_dims != 2:
-            raise Exception("Currently the fenicsx-adapter only supports 2D cases")
-
-        if self._fenicsx_dims != self._participant.get_mesh_dimensions(self._config.get_coupling_mesh_name()):
-            raise Exception("Dimension of preCICE setup and FEniCSx do not match")
-
-        if self._participant.requires_initial_data():
-            if not write_function:
+        for c_mesh in coupling_meshes:
+            write_function_space, write_function = None, None
+            # [2]: write object!
+            write_object = coupling_meshes[c_mesh][2]
+            if isinstance(write_object, fem.Function):  # precice.initialize_data() will be called using this Function
+                write_function_space = write_object.function_space
+                write_function = write_object
+            # preCICE will use default zero values for initialization.
+            elif isinstance(write_object, fem.FunctionSpace):
+                write_function_space = write_object
+                write_function = None
+            elif write_object is None:
+                pass
+            else:
+                raise Exception(
+                    f"Given write object of {c_mesh} is neither of type dolfinx.functions.function.Function or dolfinx.functions.functionspace.FunctionSpace")
```

Comment:
This line has 159 characters and is clearly too long. But autopep8 passes. Let's not worry about this here but @NiklasVin: Can you please open an issue? I assume this is some kind of bug with autopep8 or our pipeline (not dramatic but we should document it).